enriched-span-constants/src/main/java/org/hypertrace/traceenricher/enrichedspan/constants/utils/EnrichedSpanUtils.java

```diff
@@ -301,12 +317,74 @@ public static String getUserAgent(Event event) {
     return Optional.empty();
   }
 
+  /**
+   * This method returns an Optional which, if present, contains the full URL, including the protocol and host name.
+   * If the tags in the raw span do not contain the full URL, it tries to construct a full URL using the protocol
+   * ("http") and host name.
+   */
   public static Optional<String> getFullHttpUrl(Event event) {
-    if (event.getHttp() != null && event.getHttp().getRequest() != null) {
-      return Optional.ofNullable(event.getHttp().getRequest().getUrl());
+    Optional<String> optionalUrl = Optional.ofNullable(event.getHttp())
+            .map(org.hypertrace.core.datamodel.eventfields.http.Http::getRequest)
+            .map(Request::getUrl);
+
+    if (optionalUrl.isEmpty() && event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
+      // iterate on all URL related attribute keys and find / create one with protocol, host header and path
+      Map<String, AttributeValue> attributesMap = event.getAttributes().getAttributeMap();
+      for (String urlAttribute : FULL_URL_ATTRIBUTES) {
+        if (attributesMap.containsKey(urlAttribute)) {
+          String url = attributesMap.get(urlAttribute).getValue();
+          if (url != null && url.toLowerCase().startsWith(HTTP_PREFIX)) {
+            optionalUrl = Optional.of(url);
+          } else {
+            optionalUrl = constructHttpUrlFromHostName(event);
```

Comment:
The more I think about it, I'm leaning towards not doing this :) For observability, we want to show what was there in the span, not what we could construct.

Reply:
@buchi-busireddy I think this would fix https://github.com/hypertrace/hypertrace-ui/issues/283

Reply:
@skjindal93 Discussed this with @buchi-busireddy and concluded that a better way to fix this issue would be to remove the valid URL check in span-normalizer(which requires that it should be a full URL and not just the path): https://github.com/hypertrace/span-normalizer/blob/main/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java#L372 . 
We need to ensure that no one assumes that it's a full URL and if so, we need to fix those cases first.